zebra-state/src/service.rs

```diff
@@ -73,14 +73,30 @@ impl StateService {
     /// in RFC0005.
     ///
     /// [1]: https://zebra.zfnd.org/dev/rfcs/0005-state-updates.html#committing-non-finalized-blocks
-    #[instrument(skip(self, new))]
-    fn queue_and_commit_non_finalized_blocks(&mut self, new: QueuedBlock) {
-        let parent_hash = new.block.header.previous_block_hash;
+    #[instrument(skip(self, block))]
+    fn queue_and_commit_non_finalized_blocks(
+        &mut self,
+        block: Arc<Block>,
+    ) -> broadcast::Receiver<Result<block::Hash, CloneError>> {
+        let hash = block.hash();
+        let parent_hash = block.header.previous_block_hash;
+
+        if self.contains_committed_block(&block) {
+            let (rsp_tx, rsp_rx) = broadcast::channel(1);
+            let _ = rsp_tx.send(Ok(hash));
+            return rsp_rx;
+        }
 
-        self.queued_blocks.queue(new);
+        let rsp_rx = if let Some(queued_block) = self.queued_blocks.get(&hash) {
+            queued_block.rsp_tx.subscribe()
```

Comment:
if there are two pending blocks, return an error for the older pending duplicate block, and replace the older block with the newer block